src/main/java/org/csanchez/jenkins/plugins/kubernetes/ConfigUtils.java

```diff
@@ -0,0 +1,37 @@
+package org.csanchez.jenkins.plugins.kubernetes;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class ConfigUtils {
+	private static final Logger LOGGER = Logger.getLogger(ConfigUtils.class.getName());
+	public static final String KUBERNETES_NAMESPACE_PATH = "/var/run/secrets/kubernetes.io/serviceaccount/namespace";
+
+	public static String getNamespace(String namespace) {
+		LOGGER.log(Level.FINEST,
+				"getNamespace called with "+namespace);
+		if (namespace != null && !namespace.equals("")) {
```

Comment:
this `if` shouldn't be here or the filesystem namespace is never read when a default namespace is passed